core/src/display_object/movie_clip.rs

```diff
@@ -1189,16 +1189,19 @@ impl<'gc> MovieClip<'gc> {
                     dispatch_removed_event(prev_child, context);
                 }
 
-                if let Avm2Value::Object(mut p) = self.object2() {
-                    if let Avm2Value::Object(c) = child.object2() {
-                        let name = Avm2QName::new(Avm2Namespace::public(), child.name());
-                        let mut activation = Avm2Activation::from_nothing(context.reborrow());
-                        if let Err(e) = p.init_property(&name.into(), c.into(), &mut activation) {
-                            log::error!(
-                                "Got error when setting AVM2 child named \"{}\": {}",
-                                &child.name(),
-                                e
-                            );
+                if !child.base().has_default_name() {
```

Comment:
As far as I understand, there's no a real need for `DisplayObjectFlags::HAS_DEFAULT_NAME`. Instead we can check whether `child.set_name()` wasn't called a few lines above (i.e. `place_object.name.is_none()`).

Reply:
I don't think adobe animate will let you have a present but empty name (an empty field in the editor turns into a non-present name) - but if it did, I think we should assign a default one.